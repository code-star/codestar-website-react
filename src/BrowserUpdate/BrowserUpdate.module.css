.browserUpdate {
  background-color: white;
  border: 0.5em solid red;
  font-size: 200%;
  font-weight: bold;
  min-height: 6em;
  margin-top: 4rem;
  padding: 1em;
  text-align: center;
}
