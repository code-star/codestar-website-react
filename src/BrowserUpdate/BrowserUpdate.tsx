import React from 'react';
import styles from './BrowserUpdate.module.css';

const BrowserUpdate = () => (
  <div className={styles.browserUpdate}>
    <p>Please upgrade your browser.</p>
    <p>
      <a href="https://browser-update.org/update-browser.html">
        More information
      </a>
    </p>
  </div>
);

export default BrowserUpdate;
