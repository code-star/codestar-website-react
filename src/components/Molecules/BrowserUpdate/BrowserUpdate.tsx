import React, { SFC, Fragment } from 'react';
import styles from './BrowserUpdate.module.css';

const BrowserUpdate: SFC = () => {
  // Very basic feature detection that excludes IE11
  const showUpdateMessage = !window.fetch;
  return showUpdateMessage ? (
    <div className={styles.browserUpdate}>
      <p>Please upgrade your browser.</p>
      <p>
        <a href="https://browser-update.org/update-browser.html">
          More information
        </a>
      </p>
    </div>
  ) : (
    <Fragment />
  );
};

export default BrowserUpdate;
