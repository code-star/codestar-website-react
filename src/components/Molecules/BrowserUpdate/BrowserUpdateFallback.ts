/**
 * This implementation works even when React can't be loaded.
 */
import { TEXT } from './BrowserUpdate'; // TODO remove
// import styles from './BrowserUpdate.module.css';

/* tslint:disable */ const addBrowserUpdate = (elementId: any) => {
  const elem = document.createElement('div');
  elem.setAttribute(
    'style',
    `
    background-color: white;
    border: 0.5em solid red;
    font-size: 200%;
    font-weight: bold;
    min-height: 6em;
    margin-top: 4rem;
    padding: 1em;
    text-align: center;
  `
  );
  const textNode = document.createTextNode(TEXT);
  const aElem = document.createElement('a');
  aElem.setAttribute('href', 'https://browser-update.org/update-browser.html');
  const aTextNode = document.createTextNode('b');
  aElem.appendChild(aTextNode);
  elem.appendChild(textNode);
  elem.appendChild(aElem);

  const currentDiv = document.getElementById(elementId);
  document.body.insertBefore(elem, currentDiv);
};

export default addBrowserUpdate;
