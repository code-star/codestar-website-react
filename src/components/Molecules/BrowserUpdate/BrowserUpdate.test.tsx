import React from 'react';
import { shallow } from 'enzyme';

import BrowserUpdate from './BrowserUpdate';

const renderShallow = () => {
  return shallow(<BrowserUpdate />);
};

describe('<BrowserUpdate />', () => {
  describe('Snaphot', () => {
    it('shows message if window.fetch not defined', () => {
      const originalFetch = global.fetch;
      delete global.fetch;
      expect(global.renderToJSON(renderShallow())).toMatchSnapshot();
      global.fetch = originalFetch;
    });

    it('does not show message if window.fetch is defined', () => {
      expect(global.renderToJSON(renderShallow())).toMatchSnapshot();
    });
  });
});
