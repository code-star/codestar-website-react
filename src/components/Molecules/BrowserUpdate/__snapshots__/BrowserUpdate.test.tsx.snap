// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowserUpdate /> Snaphot does not show message if window.fetch is defined 1`] = `null`;

exports[`<BrowserUpdate /> Snaphot shows message if window.fetch not defined 1`] = `
<div
  className={undefined}
>
  <p>
    Please upgrade your browser.
  </p>
  <p>
    <a
      href="https://browser-update.org/update-browser.html"
    >
      More information
    </a>
  </p>
</div>
`;
